src/Http/WebApp/WebAppView.php

```diff
@@ -293,13 +293,37 @@ public function render(Request $request)
         $baseUrl = array_get($forum, 'data.attributes.baseUrl');
         $view->cssUrls = $this->buildCssUrls($baseUrl);
         $view->jsUrls = $this->buildJsUrls($baseUrl);
+        $this->addLinkHeaders();
 
         $view->head = implode("\n", $this->head);
         $view->foot = implode("\n", $this->foot);
 
         return $view->render();
     }
 
+    protected function addLinkHeaders()
+    {
+        $css = [
+            str_replace(public_path(), '', $this->getCss()->getFile()),
+            str_replace(public_path(), '', $this->localeCss->getFile())
+        ];
+        $js = [
+            str_replace(public_path(), '', $this->getJs()->getFile()),
+            str_replace(public_path(), '', $this->localeJs->getFile())
+        ];
+        foreach ($css as $cssUrl) {
+            if ($cssUrl != null) {
+                header('Link: <'.$cssUrl.'>; rel=preload, as=style', false);
+            }
+        }
+
+        foreach ($js as $jsUrl) {
+            if ($jsUrl != null) {
+                header('Link: <'.$jsUrl.'>; rel=preload, as=script', false);
```

Comment:
Hmm, we do not use the `header()` function, as they touch global state. These headers should instead be set on the [PSR-7 response object](http://www.php-fig.org/psr/psr-7/) that encapsulates the HTTP response. Trouble is, right now the response object is not available in this class - any quick idea, @tobscure?